managedtenants/core/addons_loader/addon.py

```diff
@@ -66,7 +66,23 @@ def name(self):
         return self.path.name
 
     def subscription_config_present(self):
-        return self.metadata.get("config") is not None
+        # If imageset is present, check for subscriptionConfig in the
+        # imageset file, otherwise check for the default subscription config
+        # in the addon metadata file.
+        if self.imageset:
+            if self.imageset.get("subscriptionConfig"):
+                return True
+        return self.metadata.get("subscriptionConfig") is not None
+
+    def get_subscription_config(self):
```

Comment:
nit: do we need the `subscription_config_present` function? it seems in any case, the `get_subscription_config` will return `None` if the config is not there, so we could really just do:

```python
def get_subscription_config(self):
  if self.imageset and self.imageset.get("subscriptionConfig"):
      return self.imageset.get("subscriptionConfig", None)
  return self.metadata.get("subscriptionConfig", None)
```

Reply:
Yep, that works